lib/Froxlor/Dns/Dns.php

```diff
@@ -418,15 +418,10 @@ private static function generateDkimEntries($domain)
 			}
 
 			// key
-			$dkim_txt .= 'k=rsa;p=' . trim(preg_replace('/-----BEGIN PUBLIC KEY-----(.+)-----END PUBLIC KEY-----/s', '$1', str_replace("\n", '', $domain['dkim_pubkey']))) . ';';
-
-			// service-type
-			if (Settings::Get('dkim.dkim_servicetype') == '1') {
-				$dkim_txt .= 's=email;';
-			}
+			$dkim_txt .= 'p=' . trim(preg_replace('/-----BEGIN PUBLIC KEY-----(.+)-----END PUBLIC KEY-----/s', '$1', str_replace("\n", '', $domain['dkim_pubkey']))) . ';';
 
 			// end-part
-			$dkim_txt .= 't=s';
+//			$dkim_txt .= 't=s';	// obsolete??!! DF8OE
```

Comment:
why?

> t=s indicates that any DKIM signature header using the "i=" tag must have the same domain value on the right-hand side of the @ sign in the "i=" tag and the "d=" tag (i= local-part@domain.com). The "i=" tag  domain must not be a subdomain of the "d=" tag. Do not include this tag if the use of a subdomain is required.

Reply:
On my server most domains do have subdomains - and I think I am not the only one. This must be selectable or not used at all.

Reply:
it's not about generally having or using subdomains, just mail related, but yes, can be left out, it's just a recommended option

Reply:
Not clearly descibed by me :/ I have many subdomeins which are email domains, too. I think we left this out before configuration will become too complicated and needs too much knowledge about DKIM structures...

Reply:
well we could just query the database if the domain has email enabled subdomains and set the `t` option accordingly

Reply:
That can be confusing... Most customers start without subdomains. If DKIM is now configured with the "t=" tag and customers add subdomains they will get non-working emails at the subdomains and I am sure they will not remember to correct the zonefile. Complicated. We could implement a warning when DKIM is in use and a subdomain is created?

Reply:
no, a warning is worthless when not seen. Just leave the `t` option then